dask/compatibility.py

```diff
@@ -56,10 +60,26 @@ def _getargspec(func):
     range = xrange
     reduce = reduce
     operator_div = operator.div
+    FileNotFoundError = IOError
 
     def _getargspec(func):
         return inspect.getargspec(func)
 
+    def gzip_decompress(b):
+        f = gzip.GzipFile(fileobj=BytesIO(b))
+        result = f.read()
```

Comment:
What is your concern about closing the gzipfile.  We're done with it and want to ensure that we flush out the final bit of data.

Zip is an odd beast; it's not quite a compression library (it doesn't manage a stream of bytes) and is more of an archival library (has lots of file-like things inside of it).  It's not clear to me how it would be handled in a system like this.  Other libraries, like Pandas, also don't support zip in any sort of automatic way.  It's hard to do so without making a few arbitrary guesses.